### PR TITLE
T02 add new todo

### PR DESCRIPTION
**Pull request by APaltrow**
Task: T-02: Add the possibility to create new item

- Created input element with the placeholder 
- No special symbols allowed
- After pressing "enter" new item should is created with inputed text, + dates
- Creation date set as the current time and the expiration default date set as next day
- Creation date is current time and date in format "26.04.2023 13:45"
- Add todo section has "+" icon next to the input
- After pressing "+" icon modal is opened where user can fill title, dates
- Created modal component
- Modal has 'cancel' and 'save' buttons
------
- Added adaptivity for desktop, laptop, tablet, mobile
- Added local storage persistance with Redux-persist

### CODE REVIEW

File: src/components/ui/DateTimePicker/DateTimePicker.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+export const DateTimePicker: FC<DateTimePickerProps> = ({
+  value,
+  title,
+  error = '',
+  isReadonly = false,
+
+  onChange,
+}) => {
+  const date = value ? formatISOTime(value).slice(0, 16) : '';
```
Comment (Yuliia143) [inline src/components/ui/DateTimePicker/DateTimePicker.tsx]: Magic numbers -it's better to create some const for 16

File: src/components/ui/TaskInput/TaskInput.module.scss [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  padding: 0px 40px;
+
+  font-size: $font_md;
+
+  fill: $clr_gray_primary;
+
+  @include with-border;
+  @include input-base;
+}
+.error {
```
Comment (Yuliia143) [inline src/components/ui/TaskInput/TaskInput.module.scss]: Follow one style of code - use indents between selectors or not

File: src/helpers/formatISOTime.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export const formatISOTime = (timestamp: string) => {
+  const date = new Date(timestamp);
+  const withTimeDifference = +date - new Date().getTimezoneOffset() * 60 * 1000;
```
Comment (Yuliia143) [inline src/helpers/formatISOTime.ts]: Magic numbers

File: src/hooks/useTodoTask.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  /* Validations */
+
+  useEffect(() => {
+    /* Validating text input */
+    const isValidText = validateTodoText(todo.text);
+    if (!isValidText) {
+      setTodoInputError('Task should not contain special characters');
+    } else {
+      setTodoInputError('');
+    }
```
Comment (Yuliia143) [inline src/hooks/useTodoTask.ts]: Probably it's better to create a function and have a good name for understanding, then you can remove comments

File: src/hooks/useTodoTask.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+      return;
+    }
+    /* Validating dates */
+
+    const isValidDate = validateDates(todo.creationDate, todo.expirationDate);
+    if (!isValidDate) {
+      setDateError('Incorect expiration date');
+    } else {
+      setDateError('');
+    }
```
Comment (Yuliia143) [inline src/hooks/useTodoTask.ts]: x2

Comment (APaltrow) [reply src/hooks/useTodoTask.ts]: Renamed helpers to more descriptive names.

File: src/hooks/useTodoTask.ts (hunk 4 above)
Comment (APaltrow) [reply src/hooks/useTodoTask.ts]: Renamed helpers to more descriptive names.

File: src/helpers/formatISOTime.ts (hunk 3 above)
Comment (APaltrow) [reply src/helpers/formatISOTime.ts]: Separated values to constants.

File: src/components/ui/DateTimePicker/DateTimePicker.tsx (hunk 1 above)
Comment (APaltrow) [reply src/components/ui/DateTimePicker/DateTimePicker.tsx]: Separated values to constants.

File: src/components/ui/TaskInput/TaskInput.module.scss (hunk 2 above)
Comment (APaltrow) [reply src/components/ui/TaskInput/TaskInput.module.scss]: Fixed indents between css selectors.

**Review by Yuliia143 — APPROVED** [review]

**Pull request merged**